### PR TITLE
Added more error handling, add more tracker instructions tests, fix duplicate connection test

### PR DESCRIPTION
**Pull request by mirpo**

### CODE REVIEW

File: test/integration/tracker-instructions.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        ])
+    })
+
+    afterEach(async () => {
+        await node1.stop()
+        await node2.stop()
+        await tracker.stop()
+    })
+
+    it('if failed to follow tracker instructions, inform tracker immediately about current status', async () => {
```
Comment (harbu) [inline test/integration/tracker-instructions.test.js]: I take slight issue with the word "immediate" in the description since in our code we actually have a timeout:
```js
_sendStatusToAllTrackers() {
        clearTimeout(this.sendStatusTimeout)
        this.sendStatusTimeout = setTimeout(() => {
            this.trackers.forEach((tracker) => this._sendStatus(tracker))
        }, this.opts.sendStatusToAllTrackersInterval)
}
```

**Review by harbu — APPROVED** [review]
LGTM

File: test/integration/tracker-instructions.test.js (hunk 1 above)
Comment (mirpo) [reply test/integration/tracker-instructions.test.js]: I'll update! :)
